### DISCUSSION TITLE
use verdaccio with network proxy

### DISCUSSION THREAD

**Discussion by webmapLee** [verdaccio/verdaccio#3287] [Q&A]
it didn't work when i use verdaccio with network proxy

my steps:
1. set proxy
```bash
   npm config set proxy http://proxy***
```
2.  install verdaccio, start verdaccio
3.  set registry
```bash
   npm config set registry http://localhost:4873
```
4. add user
  when i add user, throw errors:
![image](https://user-images.githubusercontent.com/20334500/182623714-a88d0e7e-0477-4e40-82f4-b1fba6421b75.png)
when i delete the proxy , i can add user ,but i can't install packages
then i add proxy, i still can't install packages
![image](https://user-images.githubusercontent.com/20334500/182624550-69fc8551-0495-4a50-81ea-d3bb64ecf3e3.png)

why? ,how can i do?

**Comment by jachstet-sea** [comment]
Is the proxy server between verdaccio and "the internet" or between the npm client and the verdaccio?

Ah, okay, since you refer to the verdaccio instance as "localhost", I would assume that the proxy is between verdaccio and the internet. If you run verdaccio on your local system, npm will not need a proxy to connect to it. Therefore, remove the proxy setting from the npm config.
To make verdaccio use a proxy server, please refer to https://github.com/verdaccio/verdaccio/issues/525 for example. It's mainly about setting environment variables for the env that verdaccio is running in

**Comment by webmapLee** [comment]
verdaccio and "the internet"

**Reply by webmapLee** [reply]
thank you, i 'll try  , in addition, if i want to set the storage location , should i just set the system eviroment
